### ISSUE TITLE
Empty headers are not sent

### ISSUE THREAD

**Issue by justinrlle** [sagebind/isahc#209]
When you gives empty headers to `isahc`, they're not sent to the server. Based on a bit of testing, it looks like it comes from the default behavior of curl.

# How to reproduce
Set an empty header when building a request:
```rust
let req = Request::builder()
        .uri("https://httpbin.org/get")
        .header("X-with-value", "a value which is preserved")
        .header("X-without-value", "")
        .body(()).expect("failed to build request");
let res = isahc::send(req).expect("failed to send request");
```
# What is the expected behavior
Both headers are sent, `X-wtih-value` with its value, and `X-without-value` with an empty value.
Because `https://httpbin.org/get` sends all received headers inside the response, the body of the response should contain the two headers, `X-with-value` and `X-without-value`.

# What is the current result
The body of the response does contain the header `X-with-value`, but does not contain the header `X-wtihout-value`:
```json
{
  "args": {}, 
  "headers": {
    "Accept": "*/*", 
    "Accept-Encoding": "deflate, gzip", 
    "Host": "httpbin.org", 
    "User-Agent": "curl/7.70.0-DEV isahc/0.9.5", 
    "X-Amzn-Trace-Id": "Root=1-5f1573b0-a7a7fdf43076e25ceaa9e2b0", 
    "X-With-Value": "a value which is preserved"
  }, 
  "origin": "REDACTED", 
  "url": "https://httpbin.org/get"
}
```

---

After some testing, I can tell it's also the behavior of the `curl` cli, but it can be overridden:
```shell
$ curl -H 'X-with-value: a value which is preserved' -H 'X-without-value:' https://httpbin.org/get
{
  "args": {}, 
  "headers": {
    "Accept": "*/*", 
    "Host": "httpbin.org", 
    "User-Agent": "curl/7.65.3", 
    "X-Amzn-Trace-Id": "Root=1-5f157454-942b9837318cfebb67180e01", 
    "X-With-Value": "a value which is preserved"
  }, 
  "origin": "REDACTED", 
  "url": "https://httpbin.org/get"
}
# but if we add a `-H 'X-without-value;'`, it works
$ curl -H 'X-with-value: a value which is preserved' -H 'X-without-value:' -H 'X-without-value;' https://httpbin.org/get
{
  "args": {}, 
  "headers": {
    "Accept": "*/*", 
    "Host": "httpbin.org", 
    "User-Agent": "curl/7.65.3", 
    "X-Amzn-Trace-Id": "Root=1-5f157471-2b95cde35a4027182631ca8d", 
    "X-With-Value": "a value which is preserved", 
    "X-Without-Value": ""
  }, 
  "origin": "REDACTED", 
  "url": "https://httpbin.org/get"
}
```

All of this can be reproduced and tested with [this gist](https://gist.github.com/justinrlle/732e916ff65aa6a7111f0360027f0ab3).

**Comment by sagebind** [comment]
Good find, thanks for the detailed report! I'll take a deeper look at this as soon as I am able.

**Comment by sagebind** [comment]
This is now fixed in the 0.9.6 release.

**Comment by justinrlle** [comment]
Wow, that was fast, thanks a lot!
